### PR TITLE
Added login-logout menu item

### PR DESCRIPTION
**Pull request by jeffysam6**
Added login-logout menu in the navigation which allows the user to login and logout

### CODE REVIEW

**Review by botanicvelious — CHANGES_REQUESTED** [review]
How do I get to the login/logout buttons? Also, don't commit your project.db unless you have to.

As for the password reset, I do not see the code necessary to make that part work?

Lastly, this is probably best done as an apphook on a page http://docs.django-cms.org/en/release-3.5.x/introduction/apphooks.html#apphooks-introduction

**Comment by jeffysam6** [comment]
login

@botanicvelious it is on the left menu bar

**Comment by botanicvelious** [comment]
Can you please fix the conflict so I can take a look

**Comment by jeffysam6** [comment]
how should i fix that @botanicvelious ?
the Login/Logout menu is working on my localhost.

**Comment by botanicvelious** [comment]
You can see in git that when you try and merge your code into the main repo that there is a merge conflict on projects.db

Honestly I am super bad at git but https://stackoverflow.com/questions/278081/resolving-a-git-conflict-with-binary-files may help?

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,6 @@ logs/*
 
 # Pyenv
 .python-version
+
+#to avoid conflict
+project.db
```
Comment (botanicvelious) [inline .gitignore]: This doesnt solve anything there is still a conflict. Also we want to not just ignore project.db, you can ignore it locally however with .git/info/exclude

**Comment by botanicvelious** [comment]
You cant just ignore that file as it looks like you needed to add a menu item, what you may have to do is pull the projects.db from the main repo, then make your changes again and commit

File: .gitignore (hunk 1 above)
Comment (jeffysam6) [reply .gitignore]: ahh okay
sorry my bad

**Comment by botanicvelious** [comment]
Also gitignore does nothing for an already tracked file, because git... you have to issue some other commands to ignore a tracked file, however you need to change the database so it see my comment above.

**Comment by jeffysam6** [comment]
okay i pulled project.db from main repo and made the changes again @botanicvelious

**Comment by botanicvelious** [comment]
I tried running your code with your database using

```BASH
git clone https://github.com/python-gsoc/python-blogs
git checkout -b jeffysam6-login-logout master
git pull https://github.com/jeffysam6/python-blogs.git login-logout
git checkout jeffysam6-login-logout project.db
```

However I do not see the menu item still.

**Comment by jeffysam6** [comment]
@botanicvelious  i merged the login-logout branch in my cloned master and it seems to work now

**Comment by botanicvelious** [comment]
I dont see an update on the pr

**Comment by jeffysam6** [comment]
can you try to clone my repo once again because i tried cloning myself and it showed up that login/logout menu @botanicvelious

**Comment by botanicvelious** [comment]
It still does not show up, did you not push maby?

You can try it with 
```BASH
git clone https://github.com/jeffysam6/python-blogs test
cd test
python manage.py runserver
```

**Comment by jeffysam6** [comment]
i followed the exact same steps and it is still showing. @botanicvelious

**Comment by botanicvelious** [comment]
Really odd, can you ping me on irc? My nick is botanic in #python-gsoc on @freenode

**Comment by botanicvelious** [comment]
In regards to our discussion, if you do it the way we talked then you will also not need to change projects.db

**Comment by jeffysam6** [comment]
okay i did those changes. @botanicvelious 
looks like it can merge now

**Comment by botanicvelious** [comment]
Dont add it to the menu part of the template, just add the login box's to the top of the page same as when you goto ?edit

**Comment by jeffysam6** [comment]
@botanicvelious  alright done

**Comment by botanicvelious** [comment]
Not the top right, the same as the ?edit login bar. https://i.imgur.com/vSBW46K.png

Just replace that bar with your bar when the user is not logged in and make sure when not logged in the cms login toolbar does not display.

**Comment by jeffysam6** [comment]
i created a login bar at the top but where should i pass the information entered? @botanicvelious

**Comment by botanicvelious** [comment]
The same way you were doing the login before?

**Comment by jeffysam6** [comment]
Capture

like this right?

then the entered information will pass throught login.html ? i will have to make a post request on login.html?
@botanicvelious

**Comment by botanicvelious** [comment]
That looks good to me, as for the url to post to it looks like in django its set to  

>en/ ^ ^cms_login/$ [name='cms_login'] 

or 

> accounts/ login/ [name='login'] 

but i could be wrong

**Comment by jeffysam6** [comment]
thank you @botanicvelious .
can you  check my pr

**Comment by botanicvelious** [comment]
The username and password fields are a bit hard to notice on the site, they blend in a bit too much. Also the alignment is a bit off. Lastly the cms toolbar does not show up when you login, you do not need to add a logout as that is built into the cms toolbar when you are logged in

**Comment by botanicvelious** [comment]
Also I do not see a "forgot password" link

**Comment by jeffysam6** [comment]
1) Should i change the color of the toolbar to black? 
2) After login, i'll have to replace my bar with cms bar?
3)and finally my bar should align with "Python Gsoc" on the top left?
@botanicvelious

**Comment by botanicvelious** [comment]
> 
> 
>     1. Should i change the color of the toolbar to black?

Maybe, will need to try it and see how it looks, currently, it blends in a lot.

>     2. After login, i'll have to replace my bar with cms bar?

you are overriding the cms bar, just dont do that

>        3)and finally my bar should align with "Python Gsoc" on the top left?

It doesnt need to line up with it, but the bottom of the input box is cut off

**Comment by jeffysam6** [comment]
okay @botanicvelious  i changed the color of the login bar and and used cms_toolbar when the user is logged in

**Comment by botanicvelious** [comment]
Please fix the merge conflict on project.db don't commit the database please

**Comment by jeffysam6** [comment]
done @botanicvelious

**Comment by botanicvelious** [comment]
- [ ] https://i.imgur.com/XSrOQRP.png Lost password should be on the same row as login
- [ ] https://i.imgur.com/6RZ71gD.png breaks at smaller screen
- [ ] https://i.imgur.com/sHKjFC5.png breaks worse on mobile

**Review by botanicvelious — APPROVED** [review]

**Comment by jeffysam6** [comment]
@botanicvelious  i'm sorry,
i'm not able to figure out how to do that

**Comment by botanicvelious** [comment]
What are you not able to figure out how to do?

**Comment by botanicvelious** [comment]
Also if you ever need help you can reach me on irc @freenode in the #python-gsoc channel just say botanic

**Comment by jeffysam6** [comment]
@botanicvelious done

**Comment by botanicvelious** [comment]
This branch has conflicts that must be resolved

**Comment by jeffysam6** [comment]
okay @botanicvelious

**Comment by jeffysam6** [comment]
okay @botanicvelious

**Comment by botanicvelious** [comment]
- [ ] Password field is readable text
- [ ] django cms logo top left is offset now
- [ ] Hovering over menu items makes the text change size? did you edit that css?

**Comment by jeffysam6** [comment]
@botanicvelious looks like my custom css was overlapping.
made changes to that

**Comment by botanicvelious** [comment]
https://i.imgur.com/Gwxa5yi.png When you hover over menu items they move from right to left.

**Comment by jeffysam6** [comment]
@botanicvelious  sorry i was late

**Pull request merged**